test cases/common/260 declare_dependency objects/meson.build

```diff
@@ -0,0 +1,23 @@
+# Test that declare_dependency(objects: ...) fixes issues with duplicated
+# objects in the final link line, thanks to deduplication of dependencies.
+# The commented declare_dependency() invocation using link_whole instead
+# fails thusly:
+#
+#     ar csrDT libbar.a libfoo.a.p/foo.c.o libbar.a.p/bar.c.o
+#     ar csrDT libfoo.a libfoo.a.p/foo.c.o
+#     cc  -o prog prog.p/prog.c.o -Wl,--as-needed -Wl,--no-undefined -Wl,--whole-archive -Wl,--start-group libfoo.a libbar.a -Wl,--end-group -Wl,--no-whole-archive
+#     /usr/bin/ld: libfoo.a.p/foo.c.o: in function `foo':
+#     ../foo.c:3: multiple definition of `foo'; libfoo.a.p/foo.c.o:../foo.c:3: first defined here
+
+project('Transitive declare_dependency(objects)', 'c')
+
+libfoo = static_library('foo', 'foo.c')
+#foo = declare_dependency(link_whole: libfoo)
```

Comment:
Are these comments meant to stay?

Reply:
Yes, they explain how to reproduce the error above.

Reply:
A negative failing test might be an alternative to the comments but I agree with @bonzini that the comments help to understand what problem `declare_dependency(objects: ...)` is solving.